### PR TITLE
PP-10922: Add failed recuring payments information to docs

### PR DESCRIPTION
**Pull request by NathanD-GDS**
### Context

Existing recurring payment docs doesn't include much information on what to do when a payment fails.

### Changes proposed in this pull request

This PR adds information about the `can_retry` attribute to the recurring payment docs and relevant API reference pages.

### CODE REVIEW

File: source/api_reference/payment_event_reference/index.html.md.erb [hunk 1]
```diff
@@ -112,6 +112,7 @@ After you send your request to get a payment’s events, you’ll receive a resp
...
 | `state.status`   | string          | What the payment's status was updated to.<br><br>Possible values are:<br><br><li>`created`<li>`started`<li>`submitted`<li>`capturable`<li>`success`<li>`failed`<li>`cancelled`<li>`error`</li><br>You can [read more about the payment status lifecycle](/api_reference/#payment-status-lifecycle). |
+| <nobr>`state.can_retry`| boolean         | Indicates if you can try to take another payment using the associated agreement for recurring payments.<br><br>If `can_retry` is `true`, you can use this agreement to try to take another recurring payment.<br><br>If `can_retry` is `false`, you cannot take another recurring payment with this agreement.<br><br>You can [read more about handling failed recurring payments](/recurring_payments/#handling-failed-recurring-payments).                                                                                                                                                                                        |
```
Comment (SandorArpa) [inline source/api_reference/payment_event_reference/index.html.md.erb:115]: I think it's worth adding that it's not always present. It's only present if:
- the payment was through RCP
- and the payment was declined

File: source/api_reference/search_payments_reference/index.html.md.erb [hunk 2]
```diff
@@ -173,6 +173,7 @@ After you send your request to search payments, you’ll receive a response simi
...
 | `state.finished` | boolean | Indicates whether a payment journey is finished. <br><br>A finished payment journey does not always mean the user has made a payment. For example, a user may submit their payment details but their bank rejects the payment - the payment journey is `finished` but no payment has actually been made. |
+| `state.can_retry`| boolean | Indicates if you can try to take another payment using the associated agreement for recurring payments.<br><br>If `can_retry` is `true`, you can use this agreement to try to take another recurring payment.<br><br>If `can_retry` is `false`, you cannot take another recurring payment with this agreement.<br><br>You can [read more about handling failed recurring payments](/recurring_payments/#handling-failed-recurring-payments).                                                                                                                                                                                        |
```
Comment (SandorArpa) [inline source/api_reference/search_payments_reference/index.html.md.erb:176]: Same as above

File: source/api_reference/single_payment_reference/index.html.md.erb [hunk 3]
```diff
@@ -132,6 +132,7 @@ After you send your request to get information about a payment, you’ll receive
...
 | `state.code`                                       | string          | [An API error code](/api_reference/#gov-uk-pay-api-error-codes) that explains why the payment failed.<br><br>`code` only appears if the payment failed.                                                                                                                                                                                        |
+| `state.can_retry`                                 | boolean          | Indicates if you can try to take another payment using the associated agreement for recurring payments.<br><br>If `can_retry` is `true`, you can use this agreement to try to take another recurring payment.<br><br>If `can_retry` is `false`, you cannot take another recurring payment with this agreement.<br><br>You can [read more about handling failed recurring payments](/recurring_payments/#handling-failed-recurring-payments).                                                                                                                                                                                        |
```
Comment (SandorArpa) [inline source/api_reference/single_payment_reference/index.html.md.erb:135]: Same as above

**Review by SandorArpa — CHANGES_REQUESTED** [review]

File: source/api_reference/payment_event_reference/index.html.md.erb (hunk 1 above)
Comment (NathanD-GDS) [reply source/api_reference/payment_event_reference/index.html.md.erb:115]: Really good point. I'll add that in.

**Review by SandorArpa — APPROVED** [review]
LGTM

**Pull request merged**